plugins/c9.ide.download/download.js

```diff
@@ -55,25 +82,40 @@ define(function(require, exports, module) {
             if (node.isFolder && node.path == "/")
                 downloadProject();
             else if (paths.length > 1)
-                vfs.download(paths);
+                downloadPaths(paths);
             else if (node.isFolder)
                 downloadFolder(node.path);
             else
                 downloadFile(node.path);
-            
+
         }
-        
+
         function downloadProject() {
-            vfs.download("/", info.getWorkspace().name + ".tar.gz");
+            vfs.download("/", info.getWorkspace().name + getArchiveFileExtension());
+        }
+
+        function downloadPaths(paths) {
+            vfs.download(paths, info.getWorkspace().name + getArchiveFileExtension());
```

Comment:
@nightwing OK. Let's use the same convention as the server uses if the client doesn't supply a filename: `first_path [+n].zip` where `n` is `(total_paths - 1)`. (See https://github.com/c9/core/blob/master/plugins/c9.vfs.server/download.js#L44).

It means duplicating the same logic on client and server, but never mind.